test/integration/framework/master_utils.go

```diff
@@ -190,8 +190,10 @@ func startMasterOrDie(masterConfig *master.Config, incomingServer *httptest.Serv
 	masterConfig.ExtraConfig.VersionedInformers = informers.NewSharedInformerFactory(clientset, masterConfig.GenericConfig.LoopbackClientConfig.Timeout)
 	m, err = masterConfig.Complete().New(genericapiserver.NewEmptyDelegate())
 	if err != nil {
-		closeFn()
-		klog.Fatalf("error in bringing up the master: %v", err)
+		func() {
+			defer klog.Fatalf("error in bringing up the master: %v", err)
+			closeFn()
```

Comment:
I think this is identical in behavior to the original code. the fatal log will happen if `closeFn()` returns, but if `closeFn()` panics, then the fatal log will not get called. Am I correct in understanding that the goal of this PR is to surface the original error when these tests fail?

Reply:
Restored to v1 of the patch.